community/primitive-collections/src/main/java/org/neo4j/cursor/RawCursor.java

```diff
@@ -50,11 +52,33 @@
 
     default void forAll( Consumer<T> consumer ) throws EXCEPTION
     {
+        mapForAll( Function.identity(), consumer );
+    }
+
+    default <R, E> E mapReduce( E initialValue, Function<T,R> map, BiFunction<R,E,E> reduce ) throws EXCEPTION
+    {
+        try
+        {
+            E current = initialValue;
+            while ( next() )
+            {
+                current = reduce.apply( map.apply( get() ), current );
+            }
+            return current;
+        }
+        finally
+        {
+            close();
+        }
+    }
+
+    default <E> void mapForAll( Function<T,E> function, Consumer<E> consumer ) throws EXCEPTION
+    {
```

Comment:
I don't know if you have any long term plans for this interface, but if we implemented `Spliterator` for `RawCursor`, we could give it a `stream()` method, and access the power of the streams framework.